### ISSUE TITLE
Invalid ENS names cause Ethereum transaction decoding to halt.

### ISSUE THREAD

**Issue by jack-richards** [rotki/rotki#7522]
## Problem Definition

While loading my transaction history ('History Events'), the process seems to fail. It is getting stuck when it attempts to load a transaction that minted an ENS name containing an invalid character. In this case, that character was 'ʀ'.

This is the error I receive in the notifications: 'Failed to retrieve missing events from transactions for ethereum chain: Error: The backend query task died unexpectedly: Invalid character: 'ʀ' | codepoint 640 (0x280)'.

Ignoring the NFT or excluding the 'token' when it appears in the 'Newly Detected Tokens' screen has no effect. It continuously runs into the issue.

## Logs

I took the liberty of omitting some of the information that might expose my wallet address. Hopefully, the logs remain useful.
```
[22/02/2024 05:48:07 GMT Standard Time] ERROR rotkehlchen.chain.ethereum.modules.ens.decoder Greenlet-7: Could not decode an ERC721 transfer for an ENS name transfer.
[22/02/2024 05:48:07 GMT Standard Time] DEBUG rotkehlchen.chain.ethereum.graph Greenlet-7: Querying The Graph for query{domains(first:1, where:{id:"id"}){name}}
[22/02/2024 05:48:07 GMT Standard Time] INFO gql.transport.requests >>> {"query": "{\n  domains(\n    first: 1\n    where: {id: \"id\"}\n  ) {\n    name\n  }\n}"}
[22/02/2024 05:48:08 GMT Standard Time] INFO gql.transport.requests <<< {"data":{"domains":[{"name":"ʀ.eth"}]}}
[22/02/2024 05:48:08 GMT Standard Time] DEBUG rotkehlchen.chain.ethereum.graph Greenlet-7: Got result from The Graph query
[22/02/2024 05:48:08 GMT Standard Time] DEBUG rotkehlchen.chain.ethereum.graph Greenlet-7: Querying The Graph for query{domains(first:1, where:{labelhash:"labelhash"}){name}}
[22/02/2024 05:48:08 GMT Standard Time] INFO gql.transport.requests >>> {"query": "{\n  domains(\n    first: 1\n    where: {labelhash: \"labelhash\"}\n  ) {\n    name\n  }\n}"}
[22/02/2024 05:48:08 GMT Standard Time] INFO gql.transport.requests <<< {"data":{"domains":[{"name":"ʀ.eth"}]}}
[22/02/2024 05:48:08 GMT Standard Time] DEBUG rotkehlchen.chain.ethereum.graph Greenlet-7: Got result from The Graph query
[22/02/2024 05:48:08 GMT Standard Time] ERROR rotkehlchen.greenlets.manager Greenlet with id 2583392411200: decode 152 ethereum transactions died with exception: Invalid character: 'ʀ' | codepoint 640 (0x280).
Exception Name: <class 'ens.exceptions.InvalidName'>
Exception Info: Invalid character: 'ʀ' | codepoint 640 (0x280)
Traceback:
   File "src\\gevent\\greenlet.py", line 908, in gevent._gevent_cgreenlet.Greenlet.run
  File "rotkehlchen\chain\evm\decoding\decoder.py", line 516, in get_and_decode_undecoded_transactions
  File "rotkehlchen\chain\evm\decoding\decoder.py", line 573, in decode_transaction_hashes
  File "rotkehlchen\chain\evm\decoding\decoder.py", line 635, in _get_or_decode_transaction_events
  File "rotkehlchen\chain\evm\decoding\decoder.py", line 433, in _decode_transaction
  File "rotkehlchen\chain\evm\decoding\decoder.py", line 345, in decode_by_address_rules
  File "rotkehlchen\chain\ethereum\modules\ens\decoder.py", line 117, in _decode_ens_registrar_event
  File "rotkehlchen\chain\ethereum\modules\ens\decoder.py", line 134, in _decode_name_registered
  File "rotkehlchen\chain\ethereum\modules\ens\decoder.py", line 81, in _save_hash_mappings_get_fullname
  File "ens\base_ens.py", line 60, in namehash
  File "ens\utils.py", line 240, in raw_name_to_hash
  File "ens\utils.py", line 132, in normalize_name
  File "ens\_normalization.py", line 504, in normalize_name_ensip15
```

### System Description

Operating system: Windows 10
Rotki version: v1.32.0

**Comment by yabirgb** [comment]
Thanks @jack-richards. We are going to look into this. Can you maybe join our discord and dm us the address that has triggered the issue so we can reproduce it? https://discord.rotki.com/

**Comment by yabirgb** [comment]
Actually @jack-richards if you can send me the address that triggers the error would be great. I don't seem to be able to find it just with the ens

**Comment by jack-richards** [comment]
Hi @yabirgb, I'd prefer to keep the exact ENS and transaction associated with the minting of it private if that's ok. As I don't want to dox my wallet. However, this behaviour can be replicated presumably with any address that owns an invalid ENS name such as [this one.](https://legacy.ens.domains/name/%E2%9D%AA%E2%97%95%E2%80%BF%E2%97%95%E2%9D%AB.eth/register)

Apologies if this makes it harder to tackle the issue.

**Comment by yabirgb** [comment]
Don't worry @jack-richards ! 100% understable. I belive it should be okey with the fix I've prepared that will be shipped hopefuly this week in 1.32.1. Sorry for the inconvenience and thanks for using rotki. If you keep having issues or anything else let us know

**Comment by jack-richards** [comment]
It's no problem, thanks for working to resolve the issue.
